### DISCUSSION TITLE
Scan a directory of XML files for changes.

### DISCUSSION THREAD

**Discussion by n0mn0m** [quartznet/quartznet#1044] [Q&A] [answered]
Hi,

I've got quartz setup running via DI with .NET Core 3 and I've configured a job using the XML plugin. Everything is working great! The next thing I was getting ready to try is refreshing the list of XML files based on a directory of files.

Has anybody set up the system to have the XML plugin files property refreshed if a new file lands in a directory? 

Basically trying to setup new Trigger/JobDetails via XML, but not put all the details in one giant XML file. Instead I was thinking a file per config, write the file to a directory, then have the XML plugin files property gets updated to include the new config file. This would also allow config without a system reload or db interactions.

I was looking through the docs and see a Quartz 2 `FileScan` job that's part of a nuget package in v3. Maybe I could use that to scan a path, then update the XML plugin files property?

**Comment by lahma** [comment, marked as answer]
The current behavior is quite static. Files are determined on [initialization stage](https://github.com/quartznet/quartznet/blob/621c9b8b9952e64bb0f20e476fdc67039df71ccc/src/Quartz.Plugins/Plugin/Xml/XMLSchedulingDataProcessorPlugin.cs#L127-L148) and won't be refreshed after that. The refresh behavior is achieved [in Start](https://github.com/quartznet/quartznet/blob/621c9b8b9952e64bb0f20e476fdc67039df71ccc/src/Quartz.Plugins/Plugin/Xml/XMLSchedulingDataProcessorPlugin.cs#L155-L217) where triggers are scheduled for each file. So unfortunately there's no quick way to signal new files to handle with the current structure that is in use. You could probably derive a solution from the current plugin which makes also the file scanning dynamic behavior.

**Reply by n0mn0m** [reply]
Got it. Thanks for the quick response @lahma! If I end up with an extension/plugin I'll get that shared out somewhere and link back here. Have a few paths I'm testing for dynamic configuration.

**Reply by lahma** [reply]
Great, if you find a solution you would like to contribute back I'd be glad to review a PR.
